sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppDeploymentImpl.java

```diff
@@ -213,12 +216,20 @@ private Mono<Void> uploadToStorage(File source, ResourceUploadDefinition option)
 
     @Override
     public SpringAppDeploymentImpl withJarFile(File jar) {
-        ensureSource(UserSourceType.JAR);
-        this.addDependency(
-            context -> parent().getResourceUploadUrlAsync()
-                .flatMap(option -> uploadToStorage(jar, option)
-                    .then(context.voidMono()))
-        );
+        if (service().isEnterpriseTier()) {
+            throw new UnsupportedOperationException("Enterprise tier artifact deployment not supported yet.");
```

Comment:
Does not support Enterprise Tier deployment due to the amount of code, will support it in next individual PR.